src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationsPanel.razor

```diff
@@ -9,46 +9,48 @@
+        <MudCollapse Expanded="@(_toggledButtonValue == NotQualified.WITHDRAWN)" >
+            <WithdrawForm Withdrawn="_notQualified as Withdrawn"
+                          OnRestore="RestoreAndUntoggle" OnEliminate="ResetButtonPrivateValue" />
         </MudCollapse>
-        <MudCollapse Expanded="@(_buttonValue == NotQualified.RETIRED)">
-            <RetireForm Retired="_notQualified as Retired" OnSubmission="Untoggle" />
+        <MudCollapse Expanded="@(_toggledButtonValue == NotQualified.RETIRED)">
+            <RetireForm Retired="_notQualified as Retired"
+                        OnRestore="RestoreAndUntoggle" OnEliminate="ResetButtonPrivateValue" />
         </MudCollapse>
-        <MudCollapse Expanded="@(_buttonValue == NotQualified.FINISHED_NOT_RANKED)">
-            <FinishNotRankedForm FinishedNotRanked="_notQualified as FinishedNotRanked" OnSubmission="Untoggle" />
+        <MudCollapse Expanded="@(_toggledButtonValue == NotQualified.FINISHED_NOT_RANKED)">
+            <FinishNotRankedForm FinishedNotRanked="_notQualified as FinishedNotRanked"
+                                 OnRestore="RestoreAndUntoggle" OnEliminate="ResetButtonPrivateValue" />
         </MudCollapse>
-        <MudCollapse Expanded="@(_buttonValue == NotQualified.DISQUALIFIED)">
-            <DisqualifyForm Disqualified="_notQualified as Disqualified" OnSubmission="Untoggle" />
+        <MudCollapse Expanded="@(_toggledButtonValue == NotQualified.DISQUALIFIED)">
+            <DisqualifyForm Disqualified="_notQualified as Disqualified"
+                            OnRestore="RestoreAndUntoggle" OnEliminate="ResetButtonPrivateValue" />
         </MudCollapse>
-        <MudCollapse Expanded="@(_buttonValue.Contains(NotQualified.FAILED_TO_QUALIFY))">
-            <FailedToQualifyForm FailedToQualify="_notQualified as FailedToQualify" OnSubmission="Untoggle" />
+        <MudCollapse Expanded="@(_toggledButtonValue.Contains(NotQualified.FAILED_TO_QUALIFY))">
+            <FailedToQualifyForm FailedToQualify="_notQualified as FailedToQualify"
+                                 OnRestore="RestoreAndUntoggle" OnEliminate="ResetButtonPrivateValue" />
         </MudCollapse>
     }
-
 </MudPaper>
 
 @code {
     [Inject]
     IParticipationBehind _participationBehind { get; set; } = default!;
     NotQualified? _notQualified => _participationBehind.SelectedParticipation?.NotQualified;
-    private string? _buttonValue {
-        get => _value != null ? _value : _notQualified?.EliminationCode;
-        set { _value = value; }
+    private string? _toggledButtonValue {
+        get => _eliminationCodeValue != null ? _eliminationCodeValue : _notQualified?.EliminationCode;
+        set { _eliminationCodeValue = value; }
     }
-    private string? _value;
-    private MudToggleGroup<string> toggleGroup = default!;
-    private Participation _selectedParticipation = default!;
+    private string? _eliminationCodeValue;
```

Comment:
Naming isn't still there. While technically speaking correct, it does not convey its actual purpose.  Think about why this property exists separately from `_toggleButtonValue`

Reply:
Still relevant. What's the reason for this field to exist and how can the name be improved to relay that reason?